src/IndexReader.zig

```diff
@@ -110,23 +110,47 @@ pub fn getNumSegments(self: *Self) usize {
     return self.memory_segments.value.count() + self.file_segments.value.count();
 }
 
-pub fn getAttributes(self: *Self, allocator: std.mem.Allocator) !std.StringHashMapUnmanaged(u64) {
-    var attributes: std.StringHashMapUnmanaged(u64) = .{};
-    errdefer attributes.deinit(allocator);
+pub fn getMetadata(self: *Self, allocator: std.mem.Allocator) !std.StringHashMapUnmanaged([]const u8) {
+    var metadata: std.StringHashMapUnmanaged([]const u8) = .{};
+    errdefer {
+        var iter = metadata.iterator();
+        while (iter.next()) |entry| {
+            allocator.free(entry.key_ptr.*);
+            allocator.free(entry.value_ptr.*);
+        }
+        metadata.deinit(allocator);
+    }
 
     inline for (segment_lists) |n| {
         const segments = @field(self, n);
         for (segments.value.nodes.items) |node| {
-            var iter = node.value.attributes.iterator();
+            var iter = node.value.metadata.iterator();
             while (iter.next()) |entry| {
-                try attributes.put(allocator, entry.key_ptr.*, entry.value_ptr.*);
+                const result = try metadata.getOrPut(allocator, entry.key_ptr.*);
+                if (!result.found_existing) {
+                    result.key_ptr.* = try allocator.dupe(u8, entry.key_ptr.*);
```

Comment:
Do we need to duplicate the memory? What if we just used the existing strings? The result of getMetadata is not long lived.